lib/tasks/avalon_ualbertalib.rake

```diff
@@ -66,6 +68,177 @@ namespace :avalon do
+        else
+          # mf.file_location may be outdated (i.e., moved outside Avalon)
+          # search for file in other directories
+          # log: if media doesn't exist or has been moved
+          glob = Dir.glob("#{ENV['SETTINGS__DROPBOX__PATH']}/**/#{File.basename(mf.file_location)}")
+          puts "ERROR: [%s] media file not found: [%s] for Object [%s]" % [mf.id, mf.file_location, mf.media_object_id] if glob.empty?
+          # assumption: every media file should have a unique name otherwise this fails and fails by reporting 2+ items
+          # log: if multiple files of the same name found but don't change execution path as there could be a legitemate reason
+          puts "ERROR: [%s] stored media file path outdated: multiple media files found with name [%s]" % [mf.id, File.basename(mf.file_location) ] if glob.count > 1
+          glob.each do |file|
+            FileUtils.mkdir_p(original_base_dir)
+            FileUtils.cp(file, "#{original_base_dir}/")
+            puts "INFO: [%s] stored media file path outdated [%s] -- using alternaive [%s]" % [mf.id, mf.file_location, file]
+          end
+        end
+        mf.derivatives.to_a.collect.each do |derivative|
+          derivative_base_dir = File.join(mf_base_dir,"derivatives","#{derivative.quality}_#{derivative.id}")
+          FileUtils.mkdir_p(derivative_base_dir)
+          File.open("#{derivative_base_dir}/derivative.json", 'w') { |file| file.write(derivative.to_json) }
+          # locate the derivative file and export
+          # derivative files are stored on the streaming server; UAL mounts this filesystem on the app server as well
+          # parse streaming server path to translate into a valid app server path
+          # start with the derivativeFile path and parse the last two subdirectories along with the filename 
+          if (File.exists?(tmp_path = File.join(options[:stream_base_dir], derivative.derivativeFile.split(File::SEPARATOR)[4..-1])))
+            FileUtils.cp(tmp_path, derivative_base_dir)
+          end
+        end
+        if mf.captions.has_content?
+          caption_base_dir = File.join(mf_base_dir, "captions")
+          FileUtils.mkdir_p(caption_base_dir)
+          File.open("#{caption_base_dir}/#{mf.captions.original_name}", 'w:BINARY') { |file| file.write(mf.captions.content) }
+        end
+        if mf.poster.has_content?
+          poster_base_dir = File.join(mf_base_dir,"poster")
+          FileUtils.mkdir_p(poster_base_dir)
+          File.open("#{poster_base_dir}/#{mf.poster.original_name}", 'w:BINARY') { |file| file.write(mf.poster.content) }
+        end
+        if mf.thumbnail.has_content?
+          thumbnail_base_dir = File.join(mf_base_dir,"thumbnail")
+          FileUtils.mkdir_p(thumbnail_base_dir)
+          File.open("#{thumbnail_base_dir}/#{mf.thumbnail.original_name}", 'w:BINARY') { |file| file.write(mf.thumbnail.content) }
+        end
+      end
+
     end
```

Comment:
Creates version 1 of the export package:

`bundle exec rake avalon:ualbertalib:export_contents -- --objects="/tmp/ids.txt" --dest_base="/tmp/delete_dest/" -derivatives="masterfiles/delete/`